### DISCUSSION TITLE
Chatting to oobabooga TextGen Webui with large context and turning off the router gives a connection timeout (Security/Privacy isue?)

### DISCUSSION THREAD

**Discussion by TechScorpion** [oobabooga/text-generation-webui#6230] [General]
I'm using oobabooga textgen webui **Locally** (which I updated quite recently within about the last couple of weeks. I don't see a version number in the UI though). I'm not using any APIs/extensions with it that are supposed to send data to the internet.
I know oobabooga textgen webui can give connection errors if you close the command prompt/program actually running it then check/try to do something in the web ui.

But I just found that while using oobabooga and AI model Meta-Llama-3-70B-Instruct.IQ1_M.gguf and asking in the Chat Intsruct mode a question about something with a large context (it was html/javascript code, and the context got cut off at 8K) and it was then running really slow to add each output token (on my RTX3090), turning off the router while it was still outputting text answering that question gave this error:
![oobabooba-chatting-with-long-context-then-turning-off-router-b](https://github.com/user-attachments/assets/348cbd07-9a64-46e4-b33d-7cf5f5cec755)

So it says "Error connection errored out". Why?
Why should it care about if I have the internet properly connected when it's supposed to be running all **locally**?
If I turn the router off before it starts processing that large prompt and ask it that question (with the 8K-ish context size) it starts answering it slowly again.

Is it it sending logging or other data to the internet if there is an internet connection while the prompt is being processed and if so why and what is being sent (does it depend what is in the prompt - eg. if there is html code in the prompt you are asking about and some of it is URLs does it try to load some of the URLs in it?)? How can I turn that off so that it runs all locally like it's supposed to (even if there is a connection to the internet it shouldn't be sending stuff every time I ask a question (or question with big context size?) unless I accept that).

**Comment by TechScorpion** [comment]
Since this is a serious issue if it's sending private info somewhere, is there no one that can give any help about this? How can I find out what is being sent to some server (if that is what is happening when it processes prompts when the router is on and there is a connection to the internet) and how can I prevent it sending info to somewhere online (eg.  details from chats that I didn't authorise it to send, which could have private info)? 

Is is sending some logging stuff only that don't contain any of the chat conversations and if so exactly what and why and can that be stopped?

**Comment by TheLounger** [comment]
I've never seen text-gen-webui trying to access the Internet, and I *always* run it with a firewall enabled, that shows any attempted connections. It does however, require a *local* connection, which a firewall can block as well.

Does your console output tell you anything?

**Comment by TechScorpion** [comment]
The console output isn't giving any text about the connection being turned off and the "connection errored out" message.

See attachment:

![oobabooga-stopped-processing-prompt-after-router-turned-off-after-leaving-it-longer](https://github.com/user-attachments/assets/22eeeec6-1332-4d5c-b13c-ffc132096f7e)

That's with it turned off part way through processing the prompt. 

Here's an attachment with the connection turned off sooner after entering the prompt:
![oobabooga-stopped-processing-prompt-after-router-turned-off-gen-button-re-enabled](https://github.com/user-attachments/assets/53325af5-5544-48e1-980d-3dcc8af86831)
It's not showing the "connection errored out" message there (maybe I didn't wait a couple of seconds for it to show that), but it's also stopped processing the prompt (right after generating the word "I'm") as the "Generate" button has returned instead of it showing the "Stop" button. Even though having a connection or not shouldn't affect oobabooga text gen as it's supposed to work all locally.

I was asking it a question about web page source (about date variables/timestamps). Could it be trying to load any links in that web page source - if it's processing the prompt while connected to the internet? Is is it sending anything else (like logging info) to the internet if there's a connection?

**Comment by TiagoTiago** [comment]
Do you have any extensions enabled?

Are you by any chance accessing it using the router's assigned IP instead of the loopback address, `localhost` (or `127.0.0.1` ) ?

**Reply by TechScorpion** [reply]
I'm using 127.0.0.1 (with a : and number on the end of it, that's what it says it's running on. The number after : will be different if there's already something running on the first number I think - eg. if I used stable diffusion and that put it on the original number).

The extensions it says it loads are:
Loading the extension "multimodal", 
Loading the extension "gallery"
The multi modal is that Llava thing, that I hardly use (as a lot of the time it didn't seem to work properly anyway. Maybe it depends on what model is being used. But when I tried removing it from the .bat file it didn't start properly until I put it back in).

I thought there might be some logging/telemetrics that's being sent when it's online (maybe something like Gradio?). When I use it I usually enable the "chat_buttons" option and use that though that's not a separate extension. I don't know what files to check/what code to see if something like Gradio is sending logging info to somewhere online and how I'd disable it if it is.
